module/data/item/weapon.mjs

```diff
@@ -157,6 +157,16 @@ export default class WeaponData extends ItemDataModel.mixin(
   prepareFinalData() {
     this.prepareFinalActivityData(this.parent.getRollData({ deterministic: true }));
     this.prepareFinalEquippableData();
+
+    const labels = this.parent.labels ??= {};
+    if ( this.hasRange ) {
```

Comment:
After considering this some more, I think I'm on-board with this logic. We shouldn't conflate reach and range which are not quite the same thing. That said, we do not really have any support for the vacuum this leaves behind, since the system has no concept of reach currently. How do we represent something like the Kraken's tentacle attack? Do we give its Attack Activity a range of 30 ft.?

Reply:
I think this is something I have on BF that I forgot to carry over, which is a `range.reach` field on a weapon. So for Kraken you would create a natural (or unarmed) weapon with the Reach property and enter `30` into the reach field. Adding the extra field separate from the range fields would also allow us to implement thrown weapons with reach.

Reply:
Reach is a little more complicated as it's kind of a property of the creature itself, and weapons inherit the creature's reach. This is pretty much only an issue for bugbears though, which we don't have a good way of representing anyway since they only gain the extra reach on their turn. I think your proposal should work for all reach weapons and natural weapons, and bugbears are no worse off than they were before.